traitlets/traitlets.py

```diff
@@ -1124,11 +1125,6 @@ def hold(change):
                 self.notify_change = notify_change
                 self._cross_validation_lock = False
 
-                if isinstance(notify_change, types.MethodType):
-                    # Presence of the notify_change method
-                    # on __dict__ can cause memory leaks
```

Comment:
@rmorshea can you remember the details of the memory leak mentioned here? It turns out in Python 3.6 that this line *causes* a massive memory leak (as in all your machine's memory in a few seconds, clearly an upstream bug). The pickle issue I think is better fixed by the pop above in getstate, but I don't recall enough about the memory leak to verify that it's not an issue anymore.